routing/index_road_graph.cpp

```diff
@@ -78,18 +86,20 @@ void IndexRoadGraph::GetRouteEdges(TEdgeVector & edges) const
 
   for (Segment const & segment : m_segments)
   {
-    if (IndexGraphStarter::IsFakeSegment(segment))
-      continue;
-
-    platform::CountryFile const & file = m_numMwmIds->GetFile(segment.GetMwmId());
-    MwmSet::MwmHandle const handle = m_index.GetMwmHandleByCountryFile(file);
-    if (!handle.IsAlive())
-      MYTHROW(RoutingException, ("Can't get mwm handle for", file));
-
-    auto featureId = FeatureID(handle.GetId(), segment.GetFeatureId());
-    edges.emplace_back(featureId, segment.IsForward(), segment.GetSegmentIdx(),
-                       GetJunction(segment, false /* front */),
-                       GetJunction(segment, true /* front */));
+    auto featureIdHandle = FeatureID();
+
+    if (!IndexGraphStarter::IsFakeSegment(segment))
+    {
+      platform::CountryFile const & file = m_numMwmIds->GetFile(segment.GetMwmId());
+      MwmSet::MwmHandle const handle = m_index.GetMwmHandleByCountryFile(file);
+      if (!handle.IsAlive())
```

Comment:
What's the purpose to take handle and almost immediately release it? To get mwm id? You have GetMwmIdByCountryFile function in indexer for that.

Reply:
Same in the code below (in GetEdges). The general rule: you don't need MwmHandle unless you're planning to work with the mwm contents.

Reply:
done